trace-utils/src/trace_utils.rs

```diff
@@ -103,6 +116,20 @@ pub fn coalesce_send_data(mut data: Vec<SendData>) -> Vec<SendData> {
         }
         false
     });
+    // Merge chunks with common properties. Reduces requests for agentful mode.
```

Comment:
Is it an optimization that we should keep for another PR? To stay on the safe side?

Reply:
Sort of mandatory fix, yes.